content/3_Silver/Prefix_Sums.mdx

```diff
@@ -681,6 +690,371 @@ as expected.
+#define F0R(i, a) for(int i = 0; i < (a); i++)
+#define trav(a, x) for (auto& a : x)
+
+void setIn(string s) { freopen(s.c_str(),"r",stdin); }
+void setOut(string s) { freopen(s.c_str(),"w",stdout); }
+void setIO(string s = "") {
+    ios_base::sync_with_stdio(0); cin.tie(0); cout.tie(0);
+    if (!s.empty()) { setIn(s+".in"), setOut(s+".out"); }
+}
+
+int N;
+
+int main() {
+    ios_base::sync_with_stdio(0);
+    cin.tie(0);
+
+    cin >> N;
+    vector<pair<int, int>> customers;
+    F0R(i, N) {
+        int a, b; cin >> a >> b;
+        customers.push_back({a, 1});
+        customers.push_back({b + 1, -1});
+    }
+
+    sort(customers.begin(), customers.end());
+
+    int curr = 0;
+    int l = 0;
+    vt<long long> pfx(2*N + 1); // prefix sum array
+    vt<int> arr(2*N);
+
+    F0R(i, 2*N) { // coordinate compression
+        if(i == 0) { customers[i].first = 0; }
+        else if(customers[i].first > curr) { l++; curr = customers[i].first; } // we move a pointer
+        arr[l+1] += customers[i].second;
+        // pfx[l+1] += customers[i].second; implementation without arr, merges all interval end and starts here
+    }
+
+    FOR(i, 1, 2*N+1) {
+        pfx[i] = arr[i] + pfx[i-1];
+        // pfx[i] += pfx[i-1]; implementation without arr
+    }
+
+    long long ret = 0;
+
+    F0R(i, 2*N+1) ret = max(ret, pfx[i]); // find our maximum value
+    cout << ret << "\n";
+}
+```
+
+#### Without Prefix Sums
+
+```cpp
+#include <bits/stdc++.h>
+ 
+using namespace std;
+ 
+#define ar array
+#define vt vector
+#define f first
+#define s second
+#define lb lower_bound
+#define ub upper_bound
+ 
+#define FOR(i, a, b) for(int i = (a); i < (b); i++)
+#define F0R(i, a) for(int i = 0; i < (a); i++)
+#define trav(a, x) for (auto& a : x)
+ 
+typedef pair<int,int> pi;
+typedef vt<pi> vpi;
+ 
+#define pb push_back
+#define mp make_pair
+#define sz(x) (int)x.size()
+#define beg(x) x.begin()
+#define en(x) x.end()
+#define all(x) beg(x), en(x)
+ 
+int N;
+ 
+int main() {
+    cin >> N;
+ 
+    vpi customers;
+ 
+    F0R(i, N) {
+        int a, b; cin >> a >> b;
+        customers.pb(mp(a, 1));
+        customers.pb(mp(b, -1));
+    }
+ 
+    sort(all(customers));
+ 
+    int count = 0;
+    int ret = 0;
+
+	//note that we don't use a prefix array in this problem because the coordinates are large.
+ 
+    trav(a, customers) {
+        count += a.s;
+        ret = max(ret, count);
+    }
+ 
+    cout << ret;
+}
+```
+
+</CPPSection>
+
+</LanguageSection>
+
+### 2-Dimensional
+
+<FocusProblem problem={problems.sample2[0]} />
+
+#### Point
+
+We can apply a similar concept to 2d matrices. Since we define $\texttt{prefix}[i][j]$ as the number of points between $(0,0)$ and $(i, j)$, each point $(i, j)$ is equal to the sum of itself and all the points between the rectangles formed by $(0, 0), (i, j-1)$ and $(0, 0),(i-1, j)$. However, since we are double counting the prefix sum at $(i-1, j-1)$, we subtract it from our equation. This can be expressed by
```

Comment:
I think it would help to say "Define $\texttt{prefix}[i][j]$ as the sum of the values in the rectangle between $(0,0)$ and $(i, j)$, in other words, $\sum_{i=1}^{N} \sum_{j=1}^{N} \texttt{arr}[i][j]. Each point ...$